### PR TITLE
Propose change to PR merging policy.

### PR DESCRIPTION
**Pull request by anntzer**
As discussed during the dev call.

attn @tacaswell 

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: doc/devel/coding_guide.rst [hunk 1]
```diff
@@ -63,7 +63,18 @@ PR Review guidelines
   :file:`doc/api/api_changes` and significant new features have and
   entry in :file:`doc/user/whats_new`.
 
-* Make sure the Travis, Appvyor, circle, and codecov tests are passing
+  - If a PR already has a positive review, a core developer (e.g. the first
+    reviewer, but not necessarily) may champion that PR for merging.  In order
+    to do so, they should ping all core devs both on github and on the dev
+    mailing list.  Other core devs can then either review the PR and merge or
+    reject it, or simply request that it gets a second review before being
+    merged.  If no one requires such a second review within a week, the PR can
```
Comment (ivanov) [inline doc/devel/coding_guide.rst]:
```suggestion
    merged.  If no one provides such a second review within a week, the PR can
```

Comment (anntzer) [reply doc/devel/coding_guide.rst]: actually the point is "if no one says 'I think this needs a second review, even though I cannot provide it myself'"

File: doc/devel/coding_guide.rst [hunk 2]
```diff
@@ -63,7 +63,18 @@ PR Review guidelines
   :file:`doc/api/api_changes` and significant new features have and
   entry in :file:`doc/user/whats_new`.
 
-* Make sure the Travis, Appvyor, circle, and codecov tests are passing
+  - If a PR already has a positive review, a core developer (e.g. the first
```
Comment (tacaswell) [inline doc/devel/coding_guide.rst]:
```suggestion
  * If a PR already has a positive review, a core developer (e.g. the first
```

**Comment by dstansby** [comment]
👍  but going to leave for now to give others in @matplotlib/developers a chance to take a look

File: doc/devel/coding_guide.rst (hunk 1 above)
Comment (jklymak) [reply doc/devel/coding_guide.rst]: Suggest we tag these so folks can easily find them?

Comment (anntzer) [reply doc/devel/coding_guide.rst]: sure.  you get to pick the color of the "Merge with single review?" label :)

Comment (jklymak) [reply doc/devel/coding_guide.rst]: Green like "Needs Review"?

Comment (anntzer) [reply doc/devel/coding_guide.rst]: deal, label created and PR edited

File: doc/devel/coding_guide.rst [hunk 3]
```diff
@@ -63,7 +63,19 @@ PR Review guidelines
...
+    reviewer, but not necessarily) may champion that PR for merging.  In order
+    to do so, they should ping all core devs both on github and on the dev
```
Comment (QuLogic) [inline doc/devel/coding_guide.rst]:
```suggestion
    to do so, they should ping all core devs both on Github and on the dev
```

File: doc/devel/coding_guide.rst [hunk 4]
```diff
@@ -63,7 +63,19 @@ PR Review guidelines
...
+    mailing list, and label the PR with the "Merge with single review?" label.
```
Comment (QuLogic) [inline doc/devel/coding_guide.rst]: I'm not sure about "_and_ on the dev mailing list"; I rarely see day-to-day pings (as opposed to long-term, more architectural things) happen there.

File: doc/devel/coding_guide.rst [hunk 5]
```diff
@@ -63,7 +63,19 @@ PR Review guidelines
...
+    Other core devs can then either review the PR and merge or reject it, or
+    simply request that it gets a second review before being merged.  If no one
+    asks for such a second review within a week, the PR can then be merged on
+    the basis of that single review.
+
+    One core dev should only champion one PR at a time and we should try to
```
Comment (QuLogic) [inline doc/devel/coding_guide.rst]:
```suggestion
    A core dev should only champion one PR at a time and we should try to
```
(current text implies only one dev & one PR at a time _simultaneously_)

File: doc/devel/coding_guide.rst (hunk 3 above)
Comment (anntzer) [reply doc/devel/coding_guide.rst]: capitalized throughout the file

File: doc/devel/coding_guide.rst (hunk 5 above)
Comment (anntzer) [reply doc/devel/coding_guide.rst]: thanks, fixed

File: doc/devel/coding_guide.rst (hunk 4 above)
Comment (anntzer) [reply doc/devel/coding_guide.rst]: I think it doesn't cost much to send an email.

Comment (QuLogic) [reply doc/devel/coding_guide.rst]: I mean that I don't see it happening regularly...

Comment (anntzer) [reply doc/devel/coding_guide.rst]: I'm fine either way, but let's see what @tacaswell thinks?

File: doc/devel/coding_guide.rst [hunk 6]
```diff
@@ -63,7 +63,19 @@ PR Review guidelines
...
 
-* Make sure the Travis, Appvyor, circle, and codecov tests are passing
+  - If a PR already has a positive review, a core developer (e.g. the first
+    reviewer, but not necessarily) may champion that PR for merging.  In order
+    to do so, they should ping all core devs both on GitHub and on the dev
+    mailing list, and label the PR with the "Merge with single review?" label.
+    Other core devs can then either review the PR and merge or reject it, or
+    simply request that it gets a second review before being merged.  If no one
+    asks for such a second review within a week, the PR can then be merged on
+    the basis of that single review.
```
Comment (WeatherGod) [inline doc/devel/coding_guide.rst]: So, if I am reading this correctly, does this mean that a core dev can effectively put a "hold" on merging a PR? If so, I think that should be made explicit. Will there be a label for that, or does one just need to leave a comment?

**Comment by WeatherGod** [comment]
Overall, I am fine with this idea. It would be nice if we had some sort of tool that helped us track this "champion" process, so that we could see how often this happens, who is doing the championing, and for whom. Also, on the flip side, to know how often PRs are getting "held up".

**Comment by WeatherGod** [comment]
Oh, and should we have some sort of provision about typical holidays where we might expect a significant number of devs to be away from their computers for a week or two at a time? I just want to make sure that no responses around the winter holidays isn't taken to mean a lack of objection.

File: doc/devel/coding_guide.rst (hunk 6 above)
Comment (anntzer) [reply doc/devel/coding_guide.rst]: Well, you can always reject a PR.  Or you can just say, "I don't want to reject this PR myself, but I think it is complex enough that I ask for a second positive review before it gets merged." (the "or simply request etc.").
You can probably just remove the "merge-with-single-review" label to signal that.

**Comment by anntzer** [comment]
I don't mind changing the one-week interval (originally suggested by @tacaswell) to a two-week interval (having a variable interval depending on the time of year seems overengineered).

**Comment by story645** [comment]
@dopplershift currently most PRs are merged within a week or remain open semi indefinitely, so 2 weeks might be a sweet spot? 

@WeatherGod I think most of those metrics might be available via the github api.

**Comment by jklymak** [comment]
I think the social stigma attached to being involved in a bad PR is substantial enough that we can trust folks to no abuse this.  i.e. waiting a bit longer if it is holidays. I think folks can also use their judgement about the scope of the PR, and the number of likely people interested/qualified to say anything about it.  Some obscure PGF change can probably be handled by a couple of folks without burdening a third to check and merge.  A major user facing change would presumably get more attempts at getting feedback

**Comment by NelleV** [comment]
I have the data for "time opened" per pull request, and it's a power law. Things that are open for more than one week are typically opened for more time than that. 

IMO, it would be reasonable to add that the person championing the PR can not be the author of the PR.

**Comment by ivanov** [comment]
@NelleV's last comment is hilarious given it's been open for an additional 26 days :)

**Pull request merged**